src/Perl6/Metamodel/CurriedRoleHOW.nqp

```diff
@@ -79,77 +79,94 @@ class Perl6::Metamodel::CurriedRoleHOW
         nqp::settypecheckmode($type, 2);
     }
 
-    method parameterize_roles($obj) {
-        my @pos_args;
-        nqp::push(@pos_args, $obj);
-        for @!pos_args {
-            nqp::push(@pos_args, $_);
+    # A curried role may be bound to form a role other than its origin, e.g.  a
+    # parametric role group and the candidate we're not quite ready to fetch at
+    # construction time. If we're still not ready yet whenever we need it, the
+    # binding is ourself for now. If the origin lacks a means of producing a
+    # binding, we assume it's its own by default.
+    method bind($obj) {
+        if $!binding =:= NQPMu {
+            if nqp::can($!curried_role.HOW, 'bind') {
```

Comment:
I'd rather make method `bind` mandatory on all role HOWs, except for `ConcreteRoleHOW` which is impossible candidate for currying. Then this condition and it's `else` block can be removed. `bind` on `ParametricRoleHOW` would then simply return itself.

Reply:
I thought the overhead of the identity binding for `Metamodel::ParametricRoleHOW` would be greater than the `nqp::can`s, but that doesn't seem to be the case.